service/notification/NotificationProcessor.php

```diff
@@ -74,87 +81,43 @@ class NotificationProcessor
         $this->helperData = $helperData;
         $this->dbInstance = $dbInstance;
         $this->orderAdapter = $orderAdapter;
         $this->customerThreadAdapter = $customerThreadAdapter;
         $this->logger = $logger;
         $this->context = $context;
+        $this->adyenPaymentResponse = $adyenPaymentResponse;
     }
 
     /**
-     * @return array|bool|null|object
-     */
-    public function getUnprocessedNotifications()
-    {
-        $dateStart = new \DateTime();
-        $dateStart->modify('-1 day');
-
-        $dateEnd = new \DateTime();
-        $dateEnd->modify('-1 minute');
-
-        $sql = 'SELECT * FROM ' . _DB_PREFIX_ . 'adyen_notification'
-            . ' WHERE `done` = 0'
-            . ' AND `processing` = 0'
-            . ' AND `created_at` > "' . $dateStart->format('Y-m-d H:i:s') . '"'
-            . ' AND `created_at` < "' . $dateEnd->format('Y-m-d H:i:s'). '"'
-            . ' LIMIT 100';
-
-        return $this->dbInstance->executeS($sql);
-    }
-
-    /**
-     * Update the unprocessed and not done notification to processing
-     * @param $id
-     * @return mixed
-     */
-    public function updateNotificationAsProcessing($id)
-    {
-        $sql = 'UPDATE ' . _DB_PREFIX_ . 'adyen_notification'
-            . ' SET `processing` = 1'
-            . ' WHERE `done` = 0'
-            . ' AND `processing` = 0'
-            . ' AND `entity_id` = "' . (int)$id . '"';
-        return $this->dbInstance->execute($sql);
-    }
-
-    /**
-     * Update the processed but not done notification to done
-     * @param $id
-     * @return mixed
+     * @param $unprocessedNotification
+     * @return bool
      */
-    public function updateNotificationAsDone($id)
+    public function processNotification($unprocessedNotification)
     {
-        $sql = 'UPDATE ' . _DB_PREFIX_ . 'adyen_notification'
-            . ' SET `processing` = 0, `done` = 1'
-            . ' WHERE `done` = 0'
-            . ' AND `processing` = 1'
-            . ' AND `entity_id` = "' . (int)$id . '"';
-        return $this->dbInstance->execute($sql);
-    }
+        // Add cron message to order
+        if (!$this->addMessage($unprocessedNotification)) {
+            return false;
+        }
 
-    /**
-     * Update the processed but not done notification to new
-     * @param $id
-     * @return mixed
-     */
-    public function updateNotificationAsNew($id)
-    {
-        $sql = 'UPDATE ' . _DB_PREFIX_ . 'adyen_notification'
-            . ' SET `processing` = 0, `done` = 0'
-            . ' WHERE `done` = 0'
-            . ' AND `processing` = 1'
-            . ' AND `entity_id` = "' . (int)$id . '"';
-        return $this->dbInstance->execute($sql);
+        switch ($unprocessedNotification['eventCode']) {
+            case AdyenNotification::AUTHORISATION:
+            case AdyenNotification::OFFER_CLOSED:
+                $this->adyenPaymentResponse->deletePaymentResponseByCartId($unprocessedNotification);
+                break;
```

Comment:
I would prefer to see an `if` statement instead.

Reply:
The only reason I created the switch because when we implement new and new notification types we are going to move to switch like in all the other plugins - so I just immediately moved there.

Reply:
While I understand the reasoning, we could achieve a similar structure with early returns instead. But then it becomes a matter of preference I think.

My problem with long `switch` statements is that it's harder to figure out what is being compared since the variable is only at the top of the statement. To write `if` statements you need to add the variable in it, which makes it more verbose.

I'll leave the decision to you because of the deadline :)

Reply:
Let's leave it like this now and we gonna discuss together with Angel how to refactor it in all the plugins so we can follow a standard approach :)